staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -588,7 +590,9 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 				handler = restfulGetResource(getter, exporter, reqScope)
 			}
 			handler = metrics.InstrumentRouteFunc(action.Verb, resource, subresource, handler)
-			handler = genericfilters.RestfulWithCompression(handler, a.group.Context)
+			if utilfeature.DefaultFeatureGate.Enabled(features.APIResponseCompression) {
```

Comment:
Not here. Please plumb as an arg to this method and plumb it back out.  You can set the feature flag as a value on the apiserver/pkg/server/config